plugins/trace_api_plugin/include/eosio/trace_api_plugin/common.hpp

```diff
@@ -4,18 +4,30 @@
 
 namespace eosio::trace_api_plugin {
 
-   using now_function = std::function<fc::time_point()>;
+   /**
+    * A function used to separate cooperative or external concerns from long running tasks
+    * calling code should expect that this can throw yield_exception and gracefully unwind if it does
+    * @throws yield_exception if the provided yield needs to terminate the long running process for any reason
+    */
+   using yield_function = std::function<void()>;
+
+   template<typename F, typename ...Args>
+   void call_if_set(F&& f, Args&&  ...args ) {
+      if (static_cast<bool>(f))
+         std::forward<F>(f)(std::forward<Args>(args)...);
+   }
 
```

Comment:
I think `yeild_function` should be a type that supports call if set instead of making the `yeild` invoker responsible. Something like `NoExceptDelegate`.
 
https://stackoverflow.com/questions/31602903/enforce-noexcept-on-stdfunction